llama-index-integrations/embeddings/llama-index-embeddings-nomic/llama_index/embeddings/nomic/base.py

```diff
@@ -94,35 +80,38 @@ def class_name(cls) -> str:
     def _embed(
         self, texts: List[str], task_type: Optional[str] = None
     ) -> List[List[float]]:
-        """Embed sentences using NomicAI."""
-        result = self._model.text(
+        result = nomic.embed.text(
             texts,
             model=self.model_name,
             task_type=task_type,
             dimensionality=self.dimensionality,
+            inference_mode=self.inference_mode,
```

Comment:
how does local mode actually work, since its using a global here? Is it loading the model into process each time? Or the nomic package has some kind of singleton for the local model?

Reply:
To keep the API simple, there is a single [global](https://github.com/nomic-ai/nomic/blob/f95f90d0b40b489f5b42e897626b323db56872fb/nomic/embed.py#L252) GPT4All instance used for local embeddings.

On the first use of local embeddings, the model is downloaded from gpt4all.io (if not already in `~/.cache/gpt4all` or equivalent) and loaded into memory. Subsequent calls to embed.text with the same model and device will use the model already in memory.

`nomic.embed.free_embedding_model()` can be called to free the resources associated with the local model.